Explorer/Assets/DCL/VoiceChat/OptimizedVoiceChatAudioProcessor.cs

```diff
@@ -0,0 +1,326 @@
+using UnityEngine;
+using System;
+using Unity.Burst;
+using Unity.Mathematics;
+
+namespace DCL.VoiceChat
+{
+    public class OptimizedVoiceChatAudioProcessor : IVoiceChatAudioProcessor
+    {
+        private readonly VoiceChatConfiguration configuration;
+        private readonly float[] lowPassPrevInputs;
+        private readonly float[] lowPassPrevOutputs;
+        private readonly float[] fadeInBuffer;
+        private readonly float[] tempBuffer;
+
+        private float peakLevel;
+        private bool gateIsOpen;
+        private float lastSpeechTime;
+        private int fadeInBufferIndex;
+        private bool isGateOpening;
+        private float gateOpenFadeProgress;
+        private float gateSmoothing;
+        private float targetGain;
+        private float currentGain;
+        private float peakTrackingTime;
+        private float lastLowPassCutoff;
+        private int lastLowPassSampleRate;
+        private float gateSpeed;
+        private float gateMultiplier;
+        private float effectiveThreshold;
+        private float fadeInSpeed;
+        private float bufferLookback;
+        private float preGateAttenuation;
+
+        [BurstCompile]
```

Comment:
Why is it `BurstCompile`, did you think about introducing jobs? Then collections should be native. individual methods can be burts compiled as well